app/generate/route.ts

```diff
@@ -49,26 +49,52 @@ const handleRequest = wrapTraced(async function handleRequest(url: string) {
 
 const getCommits = wrapTraced(async function getCommits(
   owner: string,
-  repo: string,
-) {
-  // Fetch the latest release from the GitHub API
-  const releaseResponse = await octokit.rest.repos.getLatestRelease({
-    owner,
-    repo,
-  });
-  const release = releaseResponse.data;
-  const since = release.published_at;
+  repo: string
+): Promise<{ commits: CommitsResponse['data']; since: string | null }> {
+  try {
+    // Fetch the latest release from the GitHub API
+    const releaseResponse = await octokit.rest.repos.getLatestRelease({
+      owner,
+      repo,
+    });
 
-  // Then fetch the corresponding commits
-  const commitResponse: CommitsResponse = await octokit.rest.repos.listCommits({
-    owner,
-    repo,
-    since: since ?? undefined,
-    per_page: 50,
-  });
+    let since: string | null = null;
+    if (releaseResponse && releaseResponse.data) {
+      since = releaseResponse.data.published_at;
+    }
+
+    // Fetch the latest commits (up to 20)
+    const commitResponse: CommitsResponse = await octokit.rest.repos.listCommits({
+      owner,
+      repo,
+      since: since ?? undefined,
+      per_page: 20,
+    });
+
+    const commits = commitResponse.data;
+
+    // If there was no release, set 'since' to the date of the oldest commit
+    if (!since && commits.length > 0) {
+      since = commits[commits.length - 1].commit.author?.date ?? null;
+    }
+
+    return { commits, since };
 
-  const commits = commitResponse.data;
-  return { commits, since };
+  } catch (error) {
+    if (error instanceof Error && 'status' in error && error.status === 404) {
+      // If there are no releases, just fetch the latest 20 commits
+      const commitResponse: CommitsResponse = await octokit.rest.repos.listCommits({
+        owner,
+        repo,
+        per_page: 20,
+      });
+      const commits = commitResponse.data;
+      const since = commits.length > 0 ? commits[commits.length - 1].commit.author?.date ?? null : null;
+      return { commits, since };
+    }
```

Comment:
small nit would be that it'd be nice to combine these together into one case. I think you could do that with somethig like

```
let releaseResponse = undefined;

try {
    // Fetch the latest release from the GitHub API
    releaseResponse = await octokit.rest.repos.getLatestRelease({ ...)});
catch (error) {
     if (!(error instanceof Error && 'status' in error && error.status === 404)) {
       throw error;
    }
}
```

... rest of the code that checks if release response is set, etc